src/lib.rs

```diff
@@ -22,7 +22,16 @@ pub struct OutputGroup {
     /// To denote the oldest utxo group, give them a sequence number of Some(0).
     pub creation_sequence: Option<u32>,
 }
-
+impl Ord for OutputGroup {
+    fn cmp(&self, other:&Self) -> Ordering{
+        self.creation_sequence.cmp(&other.creation_sequence)
+    }
+}
+impl PartialOrd for OutputGroup{
+    fn partial_cmp(&self, other:&Self) -> Option<Ordering>{
+        Some(self.cmp(other))
+    }
+}
```

Comment:
These implementations are not required and they are not being used.
This is also not useful because you might want to select an utxo based on it's value etc.,
You can remove them

Reply:
Understood. Removed custom order implementations. sort_by_key does the job. Also added a basic test for FIFO. If the test is along the right line, I shall add more cases.